osbenchmark/aggregator.py

```diff
@@ -184,33 +194,45 @@ def calculate_weighted_average(self, task_metrics: Dict[str, List[Any]], iterati
                     weighted_metrics[metric] = sum(values) / len(values)
         return weighted_metrics
 
-    # verify that all test executions have the same workload
-    def compatibility_check(self, test_store) -> None:
-        first_test_execution = test_store.find_by_test_execution_id(list(self.test_executions.keys())[0])
+    def test_execution_compatibility_check(self) -> None:
+        first_test_execution = self.test_store.find_by_test_execution_id(list(self.test_executions.keys())[0])
         workload = first_test_execution.workload
+        test_procedure = first_test_execution.test_procedure
         for id in self.test_executions.keys():
-            test_execution = test_store.find_by_test_execution_id(id)
+            test_execution = self.test_store.find_by_test_execution_id(id)
             if test_execution:
                 if test_execution.workload != workload:
                     raise ValueError(f"Incompatible workload: test {id} has workload '{test_execution.workload}' instead of '{workload}'")
+                if test_execution.test_procedure != test_procedure:
+                    raise ValueError(
+                        f"Incompatible test procedure: test {id} has test procedure '{test_execution.test_procedure}'\n"
+                        f"instead of '{test_procedure}'"
```

Comment:
Nit: It's good that we stated what's wrong but it'd also be good to include what should be done:
`f"Ensure that all test ids have the same test procedure from the same workload"`

This can be applied to both ValueErrors in line 205 and 208-209.

Reply:
Added this to all the error messages